Crypto/Cipher/ChaCha.hs

```diff
@@ -33,6 +37,19 @@ newtype State = State ScrubbedBytes
 newtype StateSimple = StateSimple ScrubbedBytes -- just ChaCha's state
     deriving (NFData)
 
+-- | Convert a 'StateSimple' from its internal architecture-dependent
+-- representation into a little-endian 'BS.ByteString'.
+--
+-- Note: anyone that can read this string, can break the security of the RNG.
+-- Treat it with as much importance as you would a password.
+toPortable :: StateSimple -> BS.ByteString
+toPortable (StateSimple st) = unsafeMapWords toLE32 st
+
+-- | Convert a 'StateSimple' from a little-endian 'BS.ByteString' into its
+-- internal architecture-dependent representation.
+fromPortable :: BS.ByteString -> StateSimple
+fromPortable bs = StateSimple $ unsafeMapWords fromLE32 bs
+
```

Comment:
fromPortable should verify input length and return a CryptoFailable StateSimple (for example with a new error CryptoError_StateSizeInvalid). This is how we do it elsewhere, for example Curve25519.publicKey.

Reply:
Done in 1686aee